### ISSUE TITLE
There's no way to return an error from MemoryBlockIterator

### ISSUE THREAD

**Issue by smoofra** [hillu/go-yara#118]
In C, a `YR_MEMORY_BLOCK_ITERATOR`  can set `last_error` from its iteration functions in order to return an error, but this doesn't seem to be accessible from go.

**Comment by hillu** [comment]
Guilty as charged. ;-) I think this was added to YARA for 4.1.0,  after I had done the iterator code. Just like `file_size`.

I have added another interface, `MemoryBlockIterator` in the [iterator-last-error](https://github.com/hillu/go-yara/tree/iterator-last-error) branch. Would you like to have a look if this works for you – and maybe contribute a test case for this? Thanks!

**Comment by hillu** [comment]
Making First, Next return an error would break the API. Perhaps for version 5.0.

**Comment by smoofra** [comment]
doesn't adding a new method to the interface also break the API?

**Comment by hillu** [comment]
> doesn't adding a new method to the interface also break the API?

No. For code using go-yara, implementing the extra interface is optional.

Even though this is not the most (Golang-)idiomatic way to signal an error, I can't think of a better alternative. Would you like to contirbute a test case?
